nav2_waypoint_follower/src/waypoint_follower.cpp

```diff
@@ -193,8 +220,14 @@ WaypointFollower::followWaypoints()
       }
     } else if (current_goal_status_ == ActionStatus::SUCCEEDED) {
       RCLCPP_INFO(
-        get_logger(), "Succeeded processing waypoint %i, "
-        "moving to next.", goal_index);
+        get_logger(), "Succeeded processing waypoint %i, processing waypoint task execution",
+        goal_index);
+      auto node = shared_from_this();
+      bool is_task_executed = waypoint_task_executor_->processAtWaypoint(
```

Comment:
We still need the logic here for is task executed is false and the stop_on_failure_ is true, then we should fill in the failed waypoint and exit the action (else, add the action msg and move on).

Reply:
On your comment from the old PR: we could have `!enabled_` return true not false, since that isn't a failure in the task processing. That way on default parameters there's no failing.

Reply:
I guess it is resolved in latest commit, but I keep this unresolved until you confirm that